lib/middleware/files.js

```diff
@@ -56,9 +59,8 @@ module.exports = function() {
         const pathAsDirectoryWithIndex = pathutils.asDirectoryIndex(
           pathutils.addTrailingSlash(pathname)
         );
-        return res.superstatic
-          .provider(req, pathAsDirectoryWithIndex)
-          .then((pathAsDirectoryWithIndexResult) => {
+        return providerResult(req, res, pathAsDirectoryWithIndex).then(
```

Comment:
Probably not worth fixing in this cl, but it would be easier to read if we could to find a way to not nest all of these callbacks.

Reply:
Yeah, it's the promise chains that make it hard to read. I started bringing in TS which would allow us to make it nicer, if you want to take a look: https://github.com/firebase/superstatic/pull/284